src/TestFramework/Coverage/CoverageChecker.php

```diff
@@ -0,0 +1,212 @@
+    ) {
+        $this->skipCoverage = $skipCoverage;
+        $this->skipInitialTests = $skipInitialTests;
+        $this->initialTestPhpOptions = $initialTestPhpOptions;
+        $this->coveragePath = $coveragePath;
+        $this->jUnitPath = $jUnitPath;
+        $this->testFrameworkAdapter = $testFrameworkAdapter;
+    }
+
+    public function checkCoverageRequirements(): void
+    {
+        if ($this->skipInitialTests && !$this->skipCoverage) {
+            throw new CoverageNotFound(sprintf(
+                'The initial test suite run is being skipped. The XML %sreports need to be '
+                . 'provided with the "--coverage" option',
+                $this->testFrameworkAdapter->hasJUnitReport() ?
+                    'and JUnit coverage '
+                    : ''
+            ));
+        }
+
+        if (!$this->skipCoverage && !$this->hasCoverageGeneratorEnabled()) {
+            throw new CoverageNotFound(<<<TXT
+Coverage needs to be generated but no code coverage generator (pcov, phpdbg or xdebug) has been detected. Please either:
+- Enable pcov and run infection again
+- Use phpdbg, e.g. `phpdbg -qrr infection`
+- Enable Xdebug and run infection again
+- Use the "--coverage" option with path to the existing coverage report
+- Enable the code generator tool for the initial test run only, e.g. with `--initial-tests-php-options -d zend_extension=xdebug.so`
+TXT
+            );
+        }
+    }
+
+    public function checkCoverageExists(): void
+    {
+        $coverageIndexFilePath = Path::canonicalize(
+            $this->coveragePath . DIRECTORY_SEPARATOR . PhpUnitXmlCoverageFactory::COVERAGE_INDEX_FILE_NAME
+        );
+
+        if (!file_exists($coverageIndexFilePath)) {
+            throw new CoverageNotFound(sprintf(
+                'Could not find the file "%s". Please ensure that the XML coverage report has been '
+                . 'properly generated at the right place. If using PHPUnit for example, the option '
+                . 'for the path given is "--coverage-xml=%s"',
+                $coverageIndexFilePath,
+                $this->coveragePath . '/coverage-xml'
+            ));
+        }
+
+        if ($this->testFrameworkAdapter->hasJUnitReport() && !file_exists($this->jUnitPath)) {
+            $message = sprintf(
+                'Could not find the file "%s". Please ensure that the JUnit coverage report '
+                . 'has been properly generated at the right place.',
+                $this->jUnitPath
+            );
+
+            if ($this->testFrameworkAdapter instanceof PhpUnitAdapter) {
+                $message .= sprintf(
+                    ' The PHPUnit option for the path given is "--log-junit=%s/junit.xml"',
+                    $this->coveragePath
+                );
+            } elseif ($this->testFrameworkAdapter instanceof CodeceptionAdapter) {
+                $message .= ' The Codeception option for the path given is "--xml"';
```

Comment:
I'm pretty sure this is incorrect, @maks-rafalko could you give me the right commend for codeception?

Reply:
See 

https://github.com/infection/codeception-adapter/blob/aa4db3c5748ab4236938efc71f81b08c8f793473/src/CodeceptionAdapter.php#L162-L163

But `infection/infection` must not depend on `use Infection\TestFramework\Codeception\CodeceptionAdapter;`, never. It's not correct from dependencies POV. What `infection/infection` should know is only about `TestFrameworkAdapter` interface.

If this message is so important (which I doubt), it should be on each adapter implementation